libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/feedback/core/CarFeedbackPollProvider.kt

```diff
@@ -0,0 +1,178 @@
+package com.mapbox.androidauto.car.feedback.core
+
+import androidx.annotation.DrawableRes
+import androidx.car.app.CarContext
+import androidx.car.app.model.CarIcon
+import androidx.core.graphics.drawable.IconCompat
+import com.mapbox.androidauto.R
+import com.mapbox.androidauto.car.feedback.ui.CarFeedbackOption
+import com.mapbox.androidauto.car.feedback.ui.CarFeedbackPoll
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent
+import com.mapbox.search.analytics.FeedbackEvent.FeedbackReason
+
+open class CarFeedbackPollProvider {
```

Comment:
Each screen is going to be customizable/replaceable. So it would be better if this was slightly more generic, and each screen defined it's own feedback items. It would also be great if each screen, would be able to define the feedback poll inside the screen package.

``` kotlin
fun interface CarFeedbackPollProvider {

    // Pulled from FeedbackHelper.getFreeDriveFeedbackTypes()
    fun getFeedbackPoll(carContext: CarContext): CarFeedbackPoll?
}

class ActiveGuidanceFeedbackPoll : CarFeedbackPollProvider {
    override fun getFeedbackPoll(carContext: CarContext): CarFeedbackPoll {
...
```

Reply:
Not sure I fully understand the idea. Each screen already creates its own `CarFeedbackAction`, which accepts an arbitrary `FeedbackPoll`.

Reply:
Right now there is a single `MainContext.carFeedbackPollProvider`. Considering MainCarContext is the only place to pass parameters from the outside (for now), it makes sense to put something here.

``` kotlin
class MainCarContext {
  activeGuidanceProvider: CarFeedbackPollProvider? = null
  searchFeedbackProvider: CarFeedbackPollProvider? = null
}
```

But the screens can have their own defaults. These Screen classes are public api, and people do not need to instantiate MainCarContext in order to use one or any of these Screens.

``` kotlin
class ActiveGuidanceScreen(
  feedbackProvider = ActiveGuidanceFeedbackProvider()
...
)
```

``` kotlin
class SearchScreen(
  feedbackProvider = SearchFeedbackProvider()
...
)
```

Reply:
Currently almost all of our screens' constructors accept `MainCarContext` or one of its wrappers, like `CarActiveGuidanceCarContext`, which means you have to instantiate `MainCarContext` in order to use any of these screens. Decoupling the screens from `MainCarContext` can definitely be done in a separate PR. But in this case they will simply have `FeedbackPoll` as a parameter, no need for an extra interface.